lib/utils/ensure-single-browser-tab-only.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import BootWarning from '../components/boot-warning';
+
+const HEARTBEAT_DELAY = 1000;
+const clientId = uuidv4();
+const emptyLock = [null, -Infinity];
+const foundElectron = window.process && window.process.type;
+
+if (!foundElectron) {
+  if ('lock-acquired' !== grabSessionLock()) {
+    ReactDOM.render(<BootWarning />, document.getElementById('root'));
+    throw new Error('Simplenote can only be opened in one tab');
+  }
+
+  loop(() => {
+    switch (grabSessionLock()) {
+      case 'lock-acquired':
+        return true; // keep updating the lock and look for other sessions which may have taken it
+
+      default:
+        return false; // stop watching - the user can proceed at their own risk
+    }
+  });
+
+  window.addEventListener('unload', function(event) {
+    localStorage.removeItem('session-lock');
+  });
+}
+
+function uuidv4() {
+  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+    var r = (Math.random() * 16) | 0,
+      v = c == 'x' ? r : (r & 0x3) | 0x8;
+    return v.toString(16);
+  });
+}
```

Comment:
nice. this should also be available in one of our transient dependencies `uuid`

that is, it's great that there are no dependencies, though we already pull this in, so maybe we should just use it.

you're call. don't think either one is clearly better but wanted to point out that it's in there already

Reply:
If this code was intended to live forever I'd pull it in.